### PR TITLE
Efr32xg24 dk2601b

### PR DESCRIPTION
**Pull request by marcusfolkesson**
Add support for on-board sensors.

### CODE REVIEW

File: boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b_defconfig [hunk 1]
```diff
@@ -17,3 +17,6 @@ CONFIG_CMU_HFCLK_LFXO=y
 CONFIG_HW_STACK_PROTECTION=y
 CONFIG_PINCTRL=y
 CONFIG_REGULATOR=y
+CONFIG_SI7006=y
+CONFIG_SI7210=y
+CONFIG_VEML7700=y
```
Comment (msierszulski) [inline boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b_defconfig:22]: These `CONFIG` are automatically selected if each sensor has `status = "okay"`. There is no need to add this.
See https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/veml7700/Kconfig#L9 for reference.

File: boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts [hunk 2]
```diff
@@ -82,6 +82,24 @@
...
+	};
+
+	si7210: si7210@30 {
+		compatible = "silabs,si7210";
+		reg = <0x30>;
+		status = "okay";
+	};
+
+	si7021: si7021@40 {
+		compatible = "silabs,si7006";
```
Comment (msierszulski) [inline boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts:99]: The `compatible` is different from the name of the sensor node.

**Review by msierszulski — CHANGES_REQUESTED** [review]

File: boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b_defconfig (hunk 1 above)
Comment (marcusfolkesson) [reply boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b_defconfig:22]: Got it, thank you

File: boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts [hunk 3]
```diff
@@ -82,6 +82,24 @@
 	location-sda = <GECKO_LOCATION(5) GECKO_PORT_C GECKO_PIN(5)>;
 	location-scl = <GECKO_LOCATION(5) GECKO_PORT_C GECKO_PIN(4)>;
 	status = "okay";
+
+	veml6035: veml6035@29 {
+		compatible = "vishay,veml6035", "vishay,veml7700";
```
[the remark this answers is not in the document]
Comment (marcusfolkesson) [reply boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts:87]: Correct, and I was actually quite unsure how to handle this. In the Linux kernel, the 'compatible' property contains a sorted list of strings starting with the exact name of the machine, followed by an optional list of boards it is compatible with sorted from most compatible to least. Even if the "exact" machine does not currently exists. You do not know the future and if it comes up with a driver better suitable (veml6035 in this case), we should pick that one.

How do we prefer to do in Zephyr?

File: boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts (hunk 2 above)
Comment (marcusfolkesson) [reply boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts:99]: Yes, see the comment below. I would prefer it to be

compatible = "silabs,si7021", "silabs,si7006";

File: boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts (hunk 3 above)
[the remark this answers is not in the document]
Comment (msierszulski) [reply boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts:87]: See my previous comment.

File: boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts (hunk 2 above)
Comment (marcusfolkesson) [reply boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts:99]: No one else seems to think this is a thing, so I'm going with the proposed solution.

**Review by msierszulski — APPROVED** [review]

**Review by fkokosinski — APPROVED** [review]

**Pull request merged**